hypothesis-python/src/hypothesis/internal/conjecture/minimizer.py

```diff
@@ -212,7 +212,27 @@ def minimize_as_integer(self):
         )
 
     def sort(self):
-        self.incorporate(hbytes(sorted(self.current)))
+        return self.incorporate(hbytes(sorted(self.current)))
+
+    # Sometimes data can be sorted, except that some elements need to remain
+    # in constant locations to preserve invariants that the minimizer can't be
+    # aware of. Attempt to do as much sorting as possible.
+    # This does not try to bring elements 'across' stationary elements.
+    def partial_sort(self):
+        if self.size < 2:
+            return True
+        any_sorting_done = False
+        ps = list(self.current)
```

Comment:
Given the relative cost of copying a list vs running a test function (i.e. the former is practically free) I think it would probably be clearer to just copy this each time you do the swap rather than trying to keep this in sync with self.current

Reply:
The main clarity advantage in my mind is that there's no possibility of accidentally desynchronizing the two versions of the state - by having to maintain the copy correctly there's the potential for subtle bugs. I don't think there *were* any subtle bugs in the code as was, but it's the sort of thing that can easily get introduced accidentally and it draws the eye when debugging even if that's not where the problem is.